src/app/routes/utils/withRadioSchedule/index.test.js

```diff
@@ -0,0 +1,99 @@
+import fetchMock from 'fetch-mock';
+import radioScheduleJson from '#data/hausa/bbc_hausa_radio/schedule.json';
+import withRadioSchedule from '.';
+
+const pageDataPromise = Promise.resolve({
+  json: { foo: 'bar' },
+  status: 200,
+});
+
+describe('withRadioSchedule', () => {
+  beforeEach(() => {
+    process.env.SIMORGH_BASE_URL = 'http://localhost';
+    fetchMock.restore();
+  });
+
+  describe('pageData and radioSchedule promises resolves with data', () => {
+    it('should merge radioScheduleData into pageData', async () => {
+      fetchMock.mock(
+        'http://localhost/hausa/bbc_hausa_radio/schedule.json',
+        radioScheduleJson,
+      );
+
+      const {
+        json: { radioScheduleData, foo },
+        ...rest
+      } = await withRadioSchedule(
+        pageDataPromise,
+        'hausa',
+        'http://localhost/mock-frontpage-path',
+      );
+
+      expect(radioScheduleData.length).toBeTruthy();
+      expect(foo).toBe('bar');
+      expect(rest.status).toBe(200);
+    });
+  });
+
+  describe('if either pageData or radioSchedule promise fails', () => {
+    it('should not merge radioScheduleData into pageData if radio schedule promise fails', async () => {
+      fetchMock.mock(
+        'http://localhost/hausa/bbc_hausa_radio/schedule.json',
+        404,
+      );
+
+      const {
+        json: { radioScheduleData, foo },
+        ...rest
+      } = await withRadioSchedule(
+        pageDataPromise,
+        'hausa',
+        'http://localhost/mock-frontpage-path',
+      );
+
+      expect(radioScheduleData).toBeNull();
+      expect(foo).toBe('bar');
+      expect(rest.status).toBe(200);
+    });
+
+    it('should not merge radioScheduleData into pageData if pageData promise fails', async () => {
+      fetchMock.mock(
+        'http://localhost/hausa/bbc_hausa_radio/schedule.json',
+        radioScheduleJson,
+      );
+
+      const failedPageDataPromise = Promise.resolve({
+        status: 404,
+      });
+
+      const { json, status } = await withRadioSchedule(
+        failedPageDataPromise,
+        'hausa',
+        'http://localhost/mock-frontpage-path',
+      );
+
+      expect(json).toBeUndefined();
+      expect(status).toBe(404);
+    });
+
+    it('should not merge radioScheduleData into pageData if both pageData and radioSchedule promise fails', async () => {
+      fetchMock.mock(
+        'http://localhost/hausa/bbc_hausa_radio/schedule.json',
+        404,
+      );
+
+      const failedPageDataPromise = Promise.resolve({
+        status: 404,
+      });
+
+      const { json, status } = await withRadioSchedule(
+        failedPageDataPromise,
+        'hausa',
+        'http://localhost/mock-frontpage-path',
+      );
+
+      expect(json).toBeUndefined();
+      expect(status).toBe(404);
+    });
+  });
+});
```

Comment:
We have tested so far:

- Successful data fetches (so the promise resolves)
- Unsuccessful data fetches (i.e. - the promise has resolved but the status code is not 2xx or 3xx)) 

How about if we encounter an actual error when fetching radio schedules data, so the promise is rejected? I think `fetchData` with currently throw an error that isn't caught anywhere which will give us a 500. If we want to render the page without radio schedules if the page data is fetched and radio schedules fails, we should handle this.

Reply:
I was wrong here - `fetchData` catches the error, so this will render the page without radio schedules if the promise is rejected. I'm adding a test case.